runtime/protocol/http/common/src/aws/smithy/kotlin/runtime/http/HttpBody.kt

```diff
@@ -218,3 +218,12 @@ public fun HttpBody.toSdkByteReadChannel(scope: CoroutineScope? = null): SdkByte
     is HttpBody.ChannelContent -> body.readFrom()
     is HttpBody.SourceContent -> body.readFrom().toSdkByteReadChannel(scope)
 }
+
+/**
+ * Determines if a http body is streaming type or not.
+ */
+public val HttpBody.isStreaming: Boolean
+    get() = when {
+        this is HttpBody.ChannelContent || this is HttpBody.SourceContent || this.contentLength == null -> true
+        else -> false
+    }
```

Comment:
nit: can be simplified to `(this is HttpBody.ChannelContent || this is HttpBody.SourceContent || this.contentLength == null)`

A `when` block seems overkill for a Boolean